sqlserver/metadata.csv

```diff
@@ -1,11 +1,24 @@
 metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name,curated_metric
 sqlserver.access.page_splits,gauge,,operation,second,The number of page splits per second.,-1,sql_server,page splits,
+sqlserver.ao.is_primary_replica,gauge,,,,"Returns 1 if the replica is primary, or 0 if it is a secondary replica.",0,sql_server,is primary replica,
 sqlserver.ao.ag_sync_health,gauge,,,,"Availability group synchronization health: 0 = Not healthy, 1 = Partially healthy, 2 = Healthy",0,sql_server,ag sync health,
 sqlserver.ao.replica_sync_state,gauge,,,,"Replica synchronization state: 0 = Not synchronizing, 1 = Synchronizing, 2 = Synchronized, 3 = Reverting, 4 = Initializing",0,sql_server,replica sync state,
+sqlserver.ao.log_send_queue_size,gauge,,,,"Amount of log records of the primary database that has not been sent to the secondary databases, in kilobytes (KB).",0,sql_server,log send queue size,
+sqlserver.ao.log_send_rate,gauge,,,,"Average rate at which primary replica instance sent data during last active period, in kilobytes (KB)/second.",0,sql_server,log send rate,
+sqlserver.ao.redo_queue_size,gauge,,,,"Amount of log records in the log files of the secondary replica that has not yet been redone, in kilobytes (KB).",0,sql_server,redo queue size,
+sqlserver.ao.redo_rate,gauge,,,,"Average rate at which the log records are being redone on a given secondary database, in kilobytes (KB)/second.",0,sql_server,redo rate,
+sqlserver.ao.low_water_mark_for_ghosts,gauge,,,,"A monotonically increasing number for the database indicating a low water mark used by ghost cleanup on the primary database.",0,sql_server,low water mark for ghosts,
```

Comment:
If these metrics are monotonically increasing, should the type be `monotonic_count` ? (not leading, genuinely asking).

Reply:
I think to answer this we need to decide on how we want to surface this count, because I don't think there's a right or wrong type to use here. If we use `monotonic_count` we'll surface the delta where as with `gauge` (or even `count`) we surface the value as-is. Regardless of the type we choose, the purpose of surfacing and monitoring the value of the column remains the same. Which is to see it increase over a period of time, so the value itself doesn't matter much. The nice thing with `monotonic_count` is the fact that we can more easily see the change because these numbers are quite large. After talking through it, I'm also onboard to surface this as a `monotonic_count` since I can't find any concrete reasons to show the raw value.

Reply:
I think to answer this we need to decide on how we want to surface this count and also consider the purpose of the metric. 

**Purpose**: Identify if the [ghost cleanup](https://docs.microsoft.com/en-us/sql/relational-databases/ghost-record-cleanup-process-guide?view=sql-server-ver15) is working properly by checking if this value is increasing over time.

If we use `monotonic_count` we'll surface the delta and because this value often increases slowly, the change will look constant on a graph when viewed over a long period of time. This prevents easily identifying if the ghost cleanup process is not working at a glance because a user would not see a linear graph which would indicate the job is healthy. This can be achieved with a gauge or count metric type.